internal/setup/setup.go

```diff
@@ -331,20 +330,33 @@ type yamlRelease struct {
 	V1PubKeys map[string]yamlPubKey `yaml:"v1-public-keys"`
 }
 
+type proValue string
```

Comment:
If you are going through all the trouble of defining a custom type and some constants but then define a map which is different and has to kept in sync, then we are loosing some of the value. In my opinion:
* either we keep using `proValue` in the rest of the code, including in `archive.Open` and make the constants below public,
* or we use the map instead of the switch.

The benefit of the first option is better documentation by virtue of seeing `proValue` instead of `string`. We can also define the map based on the constants although go has no way of checking whether a map is exhaustive. The last benefit is that we make it public which constants are allowed in the already public `archive.Open`.

The benefit of the map is that there is only one source of truth. This might not be too important is the map becomes `map[proValue]...` as that will make it unlikely someone will fail to define the constant.

Reply:
Are you referring to the `proArchiveInfo` map?

Removed the type, kept the constants and the map. Please take another look and let me know what you think!